internal/blockchain/fabric/fabric.go

```diff
@@ -596,8 +604,65 @@ func parseContractLocation(ctx context.Context, location fftypes.Byteable) (*Loc
 }
 
 func (f *Fabric) ValidateFFIParam(ctx context.Context, param *fftypes.FFIParam) error {
-	// TODO: Implement validation
-	return nil
+	paramDetails, err := f.parseParamDetails(ctx, param.Details)
+	if err != nil {
+		return err
+	}
+	return f.validateParamInternal(ctx, param, paramDetails)
+}
+
+func (f *Fabric) parseParamDetails(ctx context.Context, details fftypes.Byteable) (*paramDetails, error) {
+	ethParam := paramDetails{}
+	if err := json.Unmarshal(details, &ethParam); err != nil {
+		return nil, i18n.NewError(ctx, i18n.MsgContractParamInvalid, err)
+	}
+	if ethParam.Type == "" {
+		return nil, i18n.NewError(ctx, i18n.MsgContractParamInvalid, "'type' not set")
+	}
+	return &ethParam, nil
+}
+
+func (f *Fabric) validateParamInternal(ctx context.Context, param *fftypes.FFIParam, paramDetails *paramDetails) error {
+	switch {
+	case len(param.Components) > 0:
+		// object
+		if paramDetails.Type == "object" {
+			for _, childParam := range param.Components {
+				if err := f.ValidateFFIParam(ctx, childParam); err != nil {
+					return err
+				}
+			}
+			return nil
+		}
+	case strings.HasPrefix(param.Type, "byte"):
+		// byte (array)
+		if paramDetails.Type == "string" {
```

Comment:
Same comment on constant such as `FabricParamTypeString` - I also wonder how I specify bytes in Fabric?
e.g. if I had a non-UTF8 encoded binary input, how would I handle that?